### PR TITLE
publish startup as self-contained app

### PR DESCRIPTION
**Pull request by ooooolivia**
Currently startup tool is built under >=netcoreapp3.0 and the generated executable is used. In order to support netcoreapp2.x, which doesn't generate an executable, the startup project needs to be published.

### CODE REVIEW

File: src/scenarios/shared/util.py [hunk 1]
```diff
@@ -35,4 +36,9 @@ def uploadtokenpresent():
     return environ.get(UPLOAD_TOKEN_VAR) is not None
 
 def runninginlab():
-    return environ.get('PERFLAB_INLAB') is not None
\ No newline at end of file
+    return environ.get('PERFLAB_INLAB') is not None
+
+def getruntimeidentifier():
+    rid = 'win-' if sys.platform == 'win32' else 'linux-'
+    rid += 'x64' if platform.machine().endswith('64') else 'x86'
```
Comment (billwert) [inline src/scenarios/shared/util.py]: This will return `linux-x64` on our ARM64 machines.

File: src/scenarios/shared/startup.py [hunk 2]
```diff
@@ -29,11 +30,24 @@ def __init__(self):
...
+
+            startup.restore(get_packages_directory(),
+                            True,
+                            getruntimeidentifier())
+            startup.build('Release',
+                          True,
+                          get_packages_directory(),
+                          None,
+                          False,
+                          getruntimeidentifier())
```
Comment (billwert) [inline src/scenarios/shared/startup.py]: Why not just let publish do the build?

Comment (ooooolivia) [reply src/scenarios/shared/startup.py]: Sounds totally reasonable!

File: src/scenarios/shared/util.py (hunk 1 above)
Comment (ooooolivia) [reply src/scenarios/shared/util.py]: Updated.

**Comment by billwert** [comment]
Any idea why the ubuntu legs are failing?

**Comment by DrewScoggins** [comment]
Looks like we are failing to download the runtime as evidenced by the logs below. Also for some reason we are passing `release/3.1.1xx` as the channel name for every leg, which does not seem right.

```[2019/12/13 22:14:29][INFO] $ powershell.exe -NoProfile -ExecutionPolicy Bypass D:\a\1\s\CorrelationStaging\dotnet\dotnet-install.ps1 -InstallDir D:\a\1\s\CorrelationStaging\dotnet -Architecture x64 -Channel release/3.1.1xx
[2019/12/13 22:14:31][INFO] dotnet-install: Downloading link: https://dotnetcli.azureedge.net/dotnet/Sdk/3.1.101-servicing-014822/dotnet-sdk-3.1.101-servicing-014822-win-x64.zip
[2019/12/13 22:14:34][INFO] dotnet-install: Cannot download: https://dotnetcli.azureedge.net/dotnet/Sdk/3.1.101-servicing-014822/dotnet-sdk-3.1.101-servicing-014822-win-x64.zip
[2019/12/13 22:14:34][INFO] dotnet-install: Downloading legacy link: https://dotnetcli.azureedge.net/dotnet/Sdk/3.1.101-servicing-014822/dotnet-dev-win-x64.3.1.101-servicing-014822.zip
[2019/12/13 22:14:36][INFO] dotnet-install: Cannot download: https://dotnetcli.azureedge.net/dotnet/Sdk/3.1.101-servicing-014822/dotnet-dev-win-x64.3.1.101-servicing-014822.zip```

**Comment by billwert** [comment]
https://dev.azure.com/dnceng/public/_build/results?buildId=456914&view=logs is also interesting - at least some of these are hitting some weird syntax error?

**Comment by DrewScoggins** [comment]
🤦‍♂️ It's failing because we are trying to run a powershell script on Linux.

**Pull request merged**